src/mdo/java/WrapperProperties.java

```diff
@@ -45,30 +50,44 @@ class WrapperProperties extends Properties {
 
     final Supplier<Map<String, String>> getter;
     final Consumer<Properties> setter;
+    private final OrderedProperties orderedProps = new OrderedProperties();
+    private boolean initialized = false;
 
     WrapperProperties(Supplier<Map<String, String>> getter, Consumer<Properties> setter) {
         this.getter = getter;
         this.setter = setter;
     }
 
+    private synchronized void ensureInitialized() {
+        if (!initialized) {
+            Map<String, String> initial = getter.get();
+            orderedProps.putAll(initial);
+            initialized = true;
+        }
+    }
+
     @Override
     public String getProperty(String key) {
-        return getter.get().get(key);
+        ensureInitialized();
```

Comment:
why not call once in constructor, post construct, as we mimic CDI?
this seems redundant.

- https://stackoverflow.com/questions/8519187/spring-postconstruct-vs-init-method-attribute
- https://docs.oracle.com/javase/8/docs/api/javax/annotation/PostConstruct.html